tsdb/index/index.go

```diff
@@ -1797,7 +1797,12 @@ func (r *Reader) postingsForLabelMatchingV1(ctx context.Context, name string, ma
 	}
 
 	var its []Postings
+	count := 1
 	for val, offset := range e {
+		if count%100 == 0 && ctx.Err() != nil {
+			return ErrPostings(ctx.Err())
```

Comment:
I think I'm missing some bits of the context, get the ctx.Err() bit but not why we're doing count%100

Reply:
There was a slack discussion with @bboreham @pracucci about this.
I think it make sense to not do it on every iteration in such a tight loop. But YMMV

Reply:
Added const for it as way of explanation

Reply:
Just wanted to add that comment would be also nice. This is good as well.

Reply:
not tightly related to the PR:
I'm not aware of the exact overhead, but this is may be sth that we can generalize for the existing other checks in other "tight" loops:
https://github.com/prometheus/prometheus/blob/3b8b57700c469c7cde84e1d8f9d383cb8fe11ab0/web/api/v1/api.go#L884-L887
https://github.com/prometheus/prometheus/blob/3b8b57700c469c7cde84e1d8f9d383cb8fe11ab0/tsdb/head_read.go#L274-L277
(https://github.com/search?q=repo%3Aprometheus%2Fprometheus+ctx.Err%28%29&type=code for the others)
(those changes may require some benchmarks though)